CRM/Core/BAO/CustomField.php

```diff
@@ -1214,6 +1214,9 @@ public static function getDisplayValueCommon(
       case 'AdvMulti-Select':
       case 'Multi-Select':
         if (is_array($value)) {
+          // CRM-12989 fix
+          CRM_Utils_Array::formatArrayKeys($value);
+
```

Comment:
Ya, that was a miss but still formatArrayKeys() won't have an effect on those properly formatted value because the fn only process array values in typical checkbox format. But to be on safe side, I included under Checkbox condition. Thanks for spotting this :)